### ISSUE TITLE
Fix test issue with group labels

### ISSUE THREAD

**Issue by max-wittig** [python-gitlab/python-gitlab#858]
```
Traceback (most recent call last):
  File "/home//Documents/Development/python-gitlab/tools/python_test_v4.py", line 342, in <module>
    g_l = group1.labels.get("foo")
AttributeError: 'GroupLabelManager' object has no attribute 'get'
ERROR: 'python /home//Documents/Development/python-gitlab/tools/python_test_v4.py' failed
```

**Comment by sathieu** [comment]
Also, related to this,I have a `404` when trying to update an existing label.

This may be because `/projects/:id/labels/:label_id` also throws a 404.

**Comment by max-wittig** [comment]
It seems like GitLab doesn't provide an endpoint for group label get.

https://docs.gitlab.com/ce/api/group_labels.html

**Comment by sathieu** [comment]
Also there is a bug leading to `{"name": null}` json.

Because of this (when `new_data` is `{"name": "foo"}`):
```python
    # Update without ID.
    def update(self, name, new_data=None, **kwargs):
        """Update a Label on the server.

        Args:
            name: The name of the label
            **kwargs: Extra options to send to the server (e.g. sudo)
        """
        new_data = new_data or {}
        new_data["name"] = name
        super().update(id=None, new_data=new_data, **kwargs)
```

Also those bugs come from the API not being RESTful. I'm trying to address this in [gitlab-org/gitlab!17239](https://gitlab.com/gitlab-org/gitlab/merge_requests/17239)

**Comment by sathieu** [comment]
> Also there is a bug leading to `{"name": null}` json.

Proposed fix in #885.

**Comment by max-wittig** [comment]
Thanks for the fix @sathieu
